dotCMS/src/main/java/com/dotmarketing/common/db/DotDatabaseMetaData.java

```diff
@@ -0,0 +1,315 @@
+     * @throws SQLException
+     */
+    public void dropForeignKey (final Connection conn, final ForeignKey foreignKey) throws SQLException {
+        if(DbConnectionFactory.isPostgres() ||
+                DbConnectionFactory.isMsSql() ||
+                DbConnectionFactory.isOracle()) {
+
+            this.executeDropConstraint(conn, foreignKey.getForeignKeyTableName(), foreignKey.getForeignKeyName());
+        } else if (DbConnectionFactory.isMySql()) {
+            executeDropForeignKeyMySql(conn, foreignKey.getForeignKeyTableName(), foreignKey.getForeignKeyName());
+        }
+    } // dropForeignKey.
+
+    /**
+     * Drops the specified constraint associated to the given table.
+     *
+     * @param conn
+     *            - The database connection object to access the dotCMS data.
+     * @param tableName
+     *            - The name of the table whose constraint will be dropped.
+     * @param constraintName
+     *            - The name of the constraint that will be dropped.
+     * @throws SQLException
+     *             An error occurred when executing the SQL query.
+     */
+    public void executeDropConstraint(final Connection conn, final String tableName, final String constraintName) throws SQLException {
+
+        String sql = StringPool.BLANK;
+        PreparedStatement preparedStatement = null;
+
+        if(DbConnectionFactory.isMySql()) {
+            sql = (constraintName.indexOf("PRIMARY")>-1)?
+                ALTER_TABLE + tableName + " DROP PRIMARY KEY ":
+                ALTER_TABLE + tableName + " DROP INDEX " + constraintName;
+        }  else {
+            sql = ALTER_TABLE + tableName + " DROP CONSTRAINT " + constraintName;
+        }
+
+        try {
+            preparedStatement = conn.prepareStatement(sql);
+            Logger.info(this, "Executing: " + sql);
+            preparedStatement.execute();
+        }  catch (SQLException e) {
+            Logger.error(this, "Error executing: " + sql);
+            throw e;
+        } finally {
+            CloseUtils.closeQuietly(preparedStatement);
+        }
+    } // executeDropConstraint.
+
+
+    /**
+     * Drops the specified foreign key associated to the given table <b>in a
+     * MySQL database</b>.
+     *
+     * @param conn
+     *            - The database connection object to access the dotCMS data.
+     * @param tableName
+     *            - The name of the table whose foreign key will be dropped.
+     * @param constraintName
+     *            - The name of the foreign key that will be dropped.
+     * @throws SQLException
+     *             An error occurred when executing the SQL query.
+     */
+    public void executeDropForeignKeyMySql(final Connection conn, final String tableName, final String constraintName) throws SQLException {
+
+        PreparedStatement preparedStatement = null;
+        try {
+
+            preparedStatement = conn.prepareStatement(ALTER_TABLE + tableName + DROP_FOREIGN_KEY + constraintName);
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Use a variable binding mechanism to construct this query instead of concatenation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2077)